docs/solutions/integration-issues/path-shim-sandbox-recursion-prevention-2026-06-02.md

```
@@ -0,0 +1,172 @@
+  - wrapper-scripts
+  - self-name-dispatch
+plan_ref: sandbox-shim-dir-recipe
+---
+
+## Problem
+
+When using a PATH-prepended "shim directory" to transparently sandbox CLI tools, `harnx-sandbox-run` passes `PATH` through to the sandboxed child (baseline env passthrough). Birdcage resolves the command (`command[0]`) against this inherited `PATH`. If the shim directory is still first on `PATH` inside the sandbox, `harnx-sandbox-run -- node` re-resolves to the shim script → infinite recursion on the host.
+
+## Symptoms
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)